src/sql/src/plan/statement/show.rs

```diff
@@ -890,10 +890,20 @@ impl<'a> ShowColumnsSelect<'a> {
     }
 }
 
-fn simplify_names(catalog: &dyn SessionCatalog, sql: &str) -> Result<String, PlanError> {
+/// Convert a SQL statement into a form suitable for human consumption.
+fn humanize_sql(catalog: &dyn SessionCatalog, sql: &str) -> Result<String, PlanError> {
     let parsed = parse::parse(sql)?.into_element().ast;
     let (mut resolved, _) = names::resolve(catalog, parsed)?;
+
+    // Simplify names.
     let mut simplifier = NameSimplifier { catalog };
     simplifier.visit_statement_mut(&mut resolved);
+
+    // Strip internal `AS OF` syntax.
+    match &mut resolved {
+        Statement::CreateMaterializedView(stmt) => stmt.as_of = None,
+        _ => (),
+    }
+
     Ok(resolved.to_ast_string_stable())
```

Comment:
Ooc: I was surprised seeing that this doesn't use `to_ast_string_simple`, considering that we are producing a form that we are going to show to the user.